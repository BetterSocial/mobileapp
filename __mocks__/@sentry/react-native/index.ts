export const addBreadcrumb = jest.fn();
