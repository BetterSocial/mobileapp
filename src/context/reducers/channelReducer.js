import {SET_CHANNEL} from '../Types';

const channelState = {
  channel: null,
};
const channelReducer = (state = channelState, action) => {
  switch (action.type) {
    case SET_CHANNEL:
      return {
        ...state,
        channel: action.payload,
      };
    default:
      return state;
  }
};
export {channelReducer, channelState};
