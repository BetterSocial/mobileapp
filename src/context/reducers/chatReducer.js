import {SET_REPLY_TARGET} from '../Types';

const chatState = {
  replyTarget: null
};
const chatReducer = (state = chatState, action) => {
  switch (action.type) {
    case SET_REPLY_TARGET:
      return {
        ...state,
        replyTarget: action.payload
      };
    default:
      return state;
  }
};
export {chatReducer, chatState};
