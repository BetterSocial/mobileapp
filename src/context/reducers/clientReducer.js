import {SET_CHAT_CLIENT, SET_CLIENT} from '../Types';

const clientState = {
  client: null,
  chatClient: null
};
const clientReducer = (state = clientState, action) => {
  switch (action.type) {
    case SET_CLIENT:
      return {
        ...state,
        client: action.payload,
      };
    case SET_CHAT_CLIENT:
      return {
        ...state,
        chatClient: action.payload
      }
    default:
      return state;
  }
};
export {clientReducer, clientState};
