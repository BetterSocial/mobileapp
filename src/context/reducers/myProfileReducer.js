import {NAVBAR_TITLE_MY_PROFILE, SET_MY_PROFILE, SHOW_HEADER_NAVBAR} from '../Types';

const myProfileState = {
  myProfile: {},
  navbarTitle: "Who you're following",
  isShowHeader: true
};
const myProfileReducer = (state = myProfileState, action) => {
  switch (action.type) {
    case SET_MY_PROFILE:
      return {
        ...state,
        myProfile: action.payload
      };
    case NAVBAR_TITLE_MY_PROFILE:
      return {
        ...state,
        navbarTitle: action.payload
      }
    case SHOW_HEADER_NAVBAR:
      return {
        ...state,
        isShowHeader: action.payload
      }
    default:
      return state;
  }
};
export {myProfileReducer, myProfileState};
