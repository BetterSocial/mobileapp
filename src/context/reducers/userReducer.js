import {SET_DATA_HUMAN_ID, SET_DATA_IMAGE, SET_DATA_USERNAME} from '../Types';

const usersState = {
  userId: null,
  countryCode: null,
  photo: null,
  username: null,
};
const usersReducer = (state = usersState, action) => {
  switch (action.type) {
    case SET_DATA_IMAGE:
      return {
        ...state,
        photo: action.payload,
      };
    case SET_DATA_USERNAME:
      return {
        ...state,
        username: action.payload,
      };
    case SET_DATA_HUMAN_ID:
      return {
        ...state,
        userId: action.payload.userId,
        countryCode: action.payload.countryCode,
      };
    default:
      return state;
  }
};
export {usersReducer, usersState};
