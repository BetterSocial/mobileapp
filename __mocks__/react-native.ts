import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN