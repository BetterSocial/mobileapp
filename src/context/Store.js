import * as React from 'react';

import {channelReducer, channelState} from './reducers/channelReducer';
import {clientReducer, clientState} from './reducers/clientReducer';
import {discoveryReducer, discoveryState} from './reducers/discoveryReducer';
import {domainReducer, domainState} from './reducers/domainReducer';
import {feedsReducer, feedsState} from './reducers/FeedReducer';
import {followingReducer, followingState} from './reducers/followingReducer';
import {generalComponentReducer, generalComponentState} from './reducers/generalComponentReducer';
import {groupChatReducer, groupChatState} from './reducers/groupChat';
import {initialStartupReducer, initialStartupState} from './reducers/intialStartupReducer';
import {localCommunityReducer, localCommunityState} from './reducers/localComunityReducer';
import {myProfileFeedReducer, myProfileFeedState} from './reducers/myProfileFeedReducer';
import {myProfileReducer, myProfileState} from './reducers/myProfileReducer';
import {newsReducer, newsState} from './reducers/newsReducer';
import {otherProfileFeedReducer, otherProfileFeedState} from './reducers/OtherProfileFeedReducer';
import {topicsReducer, topicsState} from './reducers/topicsReducer';
import {unReadMessageReducer, unReadMessageState} from './reducers/unReadMessageReducer';
import {usersReducer, usersState} from './reducers/userReducer';
import {commentReducer, commentState} from './reducers/commentReducer';

/**
 *
 * @typedef {Object} MainContext
 * @property {import('./reducers/discoveryReducer').DiscoveryState} discovery
 */
const Store = ({children}) => {
  const rootReducer = {
    channel: React.useReducer(channelReducer, channelState),
    client: React.useReducer(clientReducer, clientState),
    discovery: React.useReducer(discoveryReducer, discoveryState),
    feeds: React.useReducer(feedsReducer, feedsState),
    following: React.useReducer(followingReducer, followingState),
    generalComponent: React.useReducer(generalComponentReducer, generalComponentState),
    groupChat: React.useReducer(groupChatReducer, groupChatState),
    localCommunity: React.useReducer(localCommunityReducer, localCommunityState),
    myProfileFeed: React.useReducer(myProfileFeedReducer, myProfileFeedState),
    otherProfileFeed: React.useReducer(otherProfileFeedReducer, otherProfileFeedState),
    news: React.useReducer(newsReducer, newsState),
    profile: React.useReducer(myProfileReducer, myProfileState),
    topics: React.useReducer(topicsReducer, topicsState),
    unReadMessage: React.useReducer(unReadMessageReducer, unReadMessageState),
    users: React.useReducer(usersReducer, usersState),
    domains: React.useReducer(domainReducer, domainState),
    initialStartup: React.useReducer(initialStartupReducer, initialStartupState),
    comments: React.useReducer(commentReducer, commentState)
  };
  return <Context.Provider value={rootReducer}>{children}</Context.Provider>;
};
export const Context = React.createContext(null);
export const useAppContext = () => React.useContext(Context);

export default Store;
