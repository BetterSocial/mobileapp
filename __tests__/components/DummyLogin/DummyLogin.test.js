import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react-native';

import DummyLogin from '../../../src/components/DevDummyLogin';
import Store, {Context} from '../../../src/context/Store';
import {myProfileState} from '../../../src/context/reducers/myProfileReducer';
import {usersState} from '../../../src/context/reducers/userReducer';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({goBack: jest.fn(), navigate: mockNavigate}),
  useRoute: () => ({
    params: {}
  })
}));

jest.mock('../../../src/utils/getstream/ClientGetStram', () => ({
  useClientGetstream: () => ({
    create: jest.fn()
  })
}));

jest.mock('react-native-config', () => ({
  ENABLE_DEV_ONLY_FEATURE: 'true'
}));

jest.mock('react-native-safe-area-context', () => {
  const inset = {top: 0, right: 0, bottom: 0, left: 0};
  return {
    SafeAreaProvider: jest.fn().mockImplementation(({children}) => children),
    SafeAreaConsumer: jest.fn().mockImplementation(({children}) => children(inset)),
    useSafeAreaInsets: jest.fn().mockImplementation(() => inset)
  };
});

jest.mock('recoil', () => ({
  useRecoilState: () => [{}, jest.fn()],
  atom: jest.fn(),
  useSetRecoilState: () => jest.fn()
}));

describe('DummyLogin should run correctly', () => {
  afterEach(cleanup);
  it('should match snapshot', () => {
    const contextValue = {
      users: [usersState, () => jest.fn()],
      profile: [myProfileState, () => jest.fn()]
    };

    const reset = jest.fn();
    const {toJSON} = renderer.create(
      <Context.Provider value={contextValue}>
        <DummyLogin resetClickTime={reset} />;
      </Context.Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('closeDummy login should run correctly', () => {
    const reset = jest.fn();
    const {getByTestId} = render(<DummyLogin resetClickTime={reset} />, {wrapper: Store});
    fireEvent.press(getByTestId('closedemo'));
    expect(reset).toHaveBeenCalled();
  });
});
