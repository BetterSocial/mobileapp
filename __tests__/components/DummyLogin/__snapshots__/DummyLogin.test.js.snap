// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyLogin should run correctly should match snapshot 1`] = `[Function]`;
